extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/user/UserService.java

```diff
@@ -392,8 +399,12 @@ public ModeledAuthenticatedUser retrieveAuthenticatedUser(AuthenticationProvider
      *     The AuthenticatedUser to retrieve the corresponding ModeledUser of.
      *
      * @return
-     *     The ModeledUser which corresponds to the given AuthenticatedUser, or
-     *     null if no such user exists.
+     *     The ModeledUser which corresponds to the given AuthenticatedUser, if
+     *     the user exists in the database.  If the user does not exist, and
+     *     the module has been configured to require JDBC authentication via
+     *     a "[db]-user-required" parameter, null will be returned.  Otherwise
```

Comment:
This now looks OK, but this feels like a separation of concerns issue to me. The `UserService` has up until now not dealt with configuration details from the `JDBCEnvironment` at all, and this function is aimed at actual retrieval of data.

What do you think about a separate function that embodies strictly the semantics of creating a skeleton, in-memory `AuthenticatedUser`? That way `retrieveAuthenticatedUser()` can truly be about retrieval and can remain independent of configuration concerns?

Reply:
Yeah, you're right.  I've taken a stab at redoing this way - creating a new `retrieveSkeletonUser()` method in the `UserService` class, and then doing all of the checks (that already happen, anyway) for configuration items in `JDBCAUthenticationProviderSrevice`.